apps/emqx_retainer/src/emqx_retainer_mnesia.erl

```diff
@@ -73,6 +76,12 @@
 topics() ->
     [emqx_topic:join(I) || I <- mnesia:dirty_all_keys(?TAB_MESSAGE)].
 
+%%--------------------------------------------------------------------
+%% Data backup
+%%--------------------------------------------------------------------
+backup_tables() ->
+    [?TAB_MESSAGE].
```

Comment:
Note: backup/restore won't work for `ram_copies` table which is a default storage type for `?TAB__MESSAGE`...
It doesn't seem to fail, but the content of the backup is empty.

Reply:
Looks like ram_copies can be backed up by adding `{ram_overrides_dump,true}]` opt:
```
mnesia:activate_checkpoint([{name, emqx_retainer_message}, {min, [emqx_retainer_message]}, {allow_remote, false}, {ram_overrides_dump,true}]).
```

Reply:
Thanks a lot for the explanation, I didn't know this before!

I thinks it won't cause problems as now the backup/restore of the retained messages are handled as following:

1. we required the user to backup the retained messages by coping the `emqx_retainer.DCD` and `emqx_retainer.DCL` dump files manually. If it is a `ram_copies` table, then one has to do `emqx eval 'mnesia:dump_tables([emqx_retainer])'` to generate a dump file.
2. convert the retained messages table using our [converter](https://github.com/emqx/emqx-data-converter), and provide a --data-files-dir option to the dir that contains dump files. See this PR for more information: https://github.com/emqx/emqx-data-converter/pull/6
3. import the generated tarball to emqx 5.6.x

Reply:
@terry-xiaoyu will open another PR to fix it.

Reply:
Hi @HJianBo, @terry-xiaoyu,
Just FYI, I've fixed backup export for RAM storage tables: https://github.com/emqx/emqx/pull/13293/commits/baa79962891ff3a597d8d9dfdf465faf46e095d6
It's now possible to export retained messages even if they are stored in ram.